ddev/src/ddev/cli/test/__init__.py

```diff
@@ -0,0 +1,220 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import click
+
+if TYPE_CHECKING:
+    from ddev.cli.application import Application
+    from ddev.integration.core import Integration
+    from ddev.utils.fs import Path
+
+
+def fix_coverage_report(report_file: Path):
+    target_dir = report_file.parent.name
+    report = report_file.read_bytes()
+
+    # Make every target's `tests` directory path unique so they don't get combined in UI
+    report = report.replace(b'"tests/', f'"{target_dir}/tests/'.encode('utf-8'))
+
+    report_file.write_bytes(report)
+
+
+@click.command(short_help='Run tests')
+@click.argument('target_spec', required=False)
+@click.argument('args', nargs=-1)
+@click.option('--lint', '-s', is_flag=True, help='Run only lint & style checks')
+@click.option('--fmt', '-fs', is_flag=True, help='Run only the code formatter')
+@click.option('--bench', '-b', is_flag=True, help='Run only benchmarks')
+@click.option('--latest', is_flag=True, help='Only verify support of new product versions')
+@click.option('--cov', '-c', 'coverage', is_flag=True, help='Measure code coverage')
+@click.option('--compat', is_flag=True, help='Check compatibility with the minimum allowed Agent version')
+@click.option('--ddtrace', is_flag=True, envvar='DDEV_TEST_ENABLE_TRACING', help='Enable tracing during test execution')
+@click.option('--memray', is_flag=True, help='Measure memory usage during test execution')
+@click.option('--recreate', '-r', is_flag=True, help='Recreate environments from scratch')
+@click.option('--list', '-l', 'list_envs', is_flag=True, help='Show available test environments')
```

Comment:
The `ddev env ls` command only displays E2E environments so there is different functionality there. Also it would be quite a big change to have the test command be a group because that would mean that either we have another subcommand to actually does the tests like `run` or the group itself executes logic based on whether or not a subcommand was invoked which is something that we don't do anywhere else except the top level ddev entry point

Reply:
I see, it's not worth it then